pkg/s3wrapper/xattr_client.go

```diff
@@ -0,0 +1,170 @@
+package s3wrapper
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/google/renameio"
+	"github.com/pkg/errors"
+	"github.com/pkg/xattr"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen --build_flags=--mod=mod -package=s3wrapper -destination=mock_xattr_client.go . XattrClient
+type XattrClient interface {
+	// IsSupported will determine if the client is supported in the current configuration
+	IsSupported() (bool, error)
+
+	SetRootDirectory(path string)
+	GetRootDirectory() string
+
+	// Set applies an extended attribute to a given path
+	// parameter: key - The key of the extended attribute.
+	// parameter: value - The value of the extended attribute.
+	// returns: an error if this fails, otherwise nil.
+	Set(tempFileName string, fileName string, key string, value string) error
+
+	// Get returns a value for an extended attribute key on a given given path.
+	// parameter: path - The path for which the key is to be fetched.
+	// parameter: key - The key of the extended attribute.
+	// returns the attribute value as a string, a boolean to indicate whether the value is ok, returns an error if there was one
+	Get(path, key string) (string, bool, error)
+
+	// List obtains a list of extended attribute keys for a given path
+	// parameter: path - The path for which extended attributes are to be fetched.
+	// returns: a list of extended attribute keys for the path or an error if this fails.
+	List(path string) ([]string, error)
+
+	// Removes an extended attribute for the file at the given path.
+	// parameter: path - The path for which the extended attribute is to be deleted.
+	// parameter: key - The key of the extended attribute.
+	// returns an error if there was an error
+	Remove(path string, key string) error
```

Comment:
Can we make it clear in the documentation that removing an attribute that doesn't exist does nothing and returns `nil`.